stubs/SQLAlchemy/sqlalchemy/orm/strategy_options.pyi

```diff
@@ -23,44 +24,116 @@ class Load(Generative, LoaderOption):
-    def noload(self, attr): ...
-    def raiseload(self, attr, sql_only: bool = ...): ...
-    def defaultload(self, attr): ...
-    def defer(self, key, raiseload: bool = ...): ...
-    def undefer(self, key): ...
-    def undefer_group(self, name): ...
-    def with_expression(self, key, expression): ...
-    def selectin_polymorphic(self, classes): ...
+    def contains_eager(loadopt, attr, alias: Incomplete | None = ...): ...
+    def load_only(loadopt, *attrs): ...
+    def joinedload(loadopt, attr, innerjoin: Incomplete | None = ...): ...
+    def subqueryload(loadopt, attr): ...
+    def selectinload(loadopt, attr): ...
+    def lazyload(loadopt, attr): ...
+    def immediateload(loadopt, attr): ...
+    def noload(loadopt, attr): ...
+    def raiseload(loadopt, attr, sql_only: bool = ...): ...
+    def defaultload(loadopt, attr): ...
+    def defer(loadopt, key, raiseload: bool = ...): ...
+    def undefer(loadopt, key): ...
+    def undefer_group(loadopt, name): ...
+    def with_expression(loadopt, key, expression): ...
+    def selectin_polymorphic(loadopt, classes): ...
 
 class _UnboundLoad(Load):
     path: Any
     local_opts: Any
     def __init__(self) -> None: ...
 
-class loader_option:
-    name: Any
-    fn: Any
-    def __call__(self, fn): ...
-
-def contains_eager(loadopt, attr, alias: Incomplete | None = ...): ...
-def load_only(loadopt, *attrs): ...
-def joinedload(loadopt, attr, innerjoin: Incomplete | None = ...): ...
-def subqueryload(loadopt, attr): ...
-def selectinload(loadopt, attr): ...
-def lazyload(loadopt, attr): ...
-def immediateload(loadopt, attr): ...
-def noload(loadopt, attr): ...
-def raiseload(loadopt, attr, sql_only: bool = ...): ...
-def defaultload(loadopt, attr): ...
-def defer(loadopt, key, raiseload: bool = ...): ...
-def undefer(loadopt, key): ...
-def undefer_group(loadopt, name): ...
-def with_expression(loadopt, key, expression): ...
-def selectin_polymorphic(loadopt, classes): ...
+###
+# The methods below are decorated with the class loader_option
+# They dynamically become instances of loader_option,
+# wich is callable with their original parameters.
+#
+# While both mypy and pyright's validation work, Pylance is unable to
+# show the parameters and return types.
+#
+# There is a workaround (define the method for Pylance, then reassign
+# an instance of loader_option to it for mypy, and add pyright+Flake8
+# suppressions), but it is too hacky and relies on some unsupported quirks.
+#
+# Asking Pylance to add support for these generic callables might be preferable.
+###
+
+_F = TypeVar("_F", bound=Callable[..., loader_option[Any]])
+
+class loader_option(Generic[_F]):
+    name: str
+    _dynamic: _F
+    fn: _F
+    __call__: _F  # Cheesy "__call__" definition to use the dynamic methods instead
+
+class _contains_eager(Protocol):
+    def __call__(self, loadopt, attr, alias: Incomplete | None = ...) -> loader_option[_contains_eager]: ...
+
+contains_eager: loader_option[_contains_eager]
```

Comment:
This is the workaround I'm mentioning in the comment above:
```suggestion
class _contains_eager(Protocol):
    def __call__(self, loadopt, attr, alias: Incomplete | None = ...) -> loader_option[_contains_eager]: ...

def contains_eager(loadopt, attr, alias: Incomplete | None = ...) -> loader_option[_contains_eager]: ...

contains_eager = loader_option[_contains_eager]()  # pyright: ignore[reportGeneralTypeIssues]  # noqa: F811
```

As you can see it's quite hacky and relies on ordering quirks from pyright and mypy. And it would only be to show more information in a specific IDE (Pylance in VSCode).